### PR TITLE
Join two leaderboards together, filter and annotate one of them with the other's values

### PR DESCRIPTION
**Pull request by thomasyu888**
In Synapse, there is currently not an "easy" way for participants to submit their write-ups to challenges.  We could have participants submit pdfs, but the thought is we want participants to leverage Synapse Projects and submit the project itself.   That being said, this tool does the following:

1.  Archives a submitted writeup (copies the project) given a `objectId` and attaches the project entity id created by the archive as a submission status annotation.
2.  Merges a writeup evaluation queue and main challenge evaluation queue by `submitterId` and adds submission status annotations of writeup entity id, archived writeup entity id for submissions on the main challenge evaluation queues

### CODE REVIEW

File: challengeutils/writeup_attacher.py [hunk 1]
```diff
@@ -1,52 +1,154 @@
...
+    Args:
+        syn: Synapse object
+        submissionid: Synapse submission objectId
+        rearchive: Boolean value to rearchive a project or not
+    """
+    # retrieve file into cache and copy it to destination
+    sub = syn.getSubmission(submissionid, downloadFile=False)
+    sub_status = syn.getSubmissionStatus(submissionid)
+    # The .get accounts for if there is no stringAnnos
+    check_if_archived = filter(lambda x: x.get("key") == "archived",
```
Comment (kdaily) [inline challengeutils/writeup_attacher.py]: should use list comprehension

File: challengeutils/writeup_attacher.py [hunk 2]
```diff
@@ -1,52 +1,154 @@
...
+                               sub_status.annotations.get('stringAnnos', []))
+    archived_entity = list(check_if_archived)
```
Comment (kdaily) [inline challengeutils/writeup_attacher.py]: if list comprehension used, then this isn't necessary

File: challengeutils/writeup_attacher.py [hunk 3]
```diff
@@ -1,52 +1,154 @@
...
+                     f"{sub.id} {sub.entityId}")
+    project_entity = synapseclient.Project(archived_name)
+    entity = syn.store(project_entity)
+    synapseutils.copy(syn, sub.entityId, entity.id)
+    return entity
+
+
+def archive_writeup(syn, submissionid, rearchive=False):
+    """
+    Archive one writeup submission
```
Comment (kdaily) [inline challengeutils/writeup_attacher.py]: this should discuss the logic of what rearchive means.

File: challengeutils/writeup_attacher.py [hunk 4]
```diff
@@ -1,52 +1,154 @@
...
+        entity = _create_archive_writeup(syn, sub)
+        archived = {"archived": entity.id}
+        sub_status = utils.update_single_submission_status(sub_status,
+                                                           archived)
+        syn.store(sub_status)
+        return entity.id
+    return archived_entity[0]['value']
+
+
+def archive_writeups(syn, evaluation, status="VALIDATED", rearchive=False):
```
Comment (kdaily) [inline challengeutils/writeup_attacher.py]: be more specific with fxn name: `archive_evaluation_writeups` or something

**Comment by thomasyu888** [comment]
I don't disagree, but the term `writeups` is what we use for the challenges.  Just trying to match the terminology.

File: challengeutils/writeup_attacher.py (hunk 4 above)
Comment (thomasyu888) [reply challengeutils/writeup_attacher.py]: What do you think about `archive_writeup_submissions`

**Comment by kdaily** [comment]
Fair to not change to 'narrative'. How about not using the term at all? The internal code doesn't need to call things by the same names that they are in use. A 'writeup' is just a Project submitted to an evaluation queue. How about calling them project functions? `archive_project_submissions`?

**Comment by kdaily** [comment]
Using `project`-named functions would maintain consistency with Synapse features - when you go to submit to a queue, it's not going to ask you to 'select a write up', rather 'select an entity' and they're going to select their `Project`.

**Comment by thomasyu888** [comment]
No longer as much value for this because of submission views.

**Pull request closed without merging**